core/station/impl/src/repositories/indexes/request_policy_resource_index.rs

```diff
@@ -248,4 +324,34 @@ mod tests {
 
         assert_eq!(policies.len(), 5);
     }
+
+    #[test]
+    fn test_find_external_canister_call_policies_by_execution_method() {
+        let repository = RequestPolicyResourceIndexRepository::default();
+        for i in 0..10 {
+            let index = RequestPolicyResourceIndex {
+                resource: Resource::ExternalCanister(ExternalCanisterResourceAction::Call(
+                    CallExternalCanisterResourceTarget {
+                        execution_method: ExecutionMethodResourceTarget::ExecutionMethod(
+                            CanisterMethod {
+                                canister_id: Principal::management_canister(),
+                                method_name: format!("method_{}", i),
+                            },
+                        ),
+                        validation_method: ValidationMethodResourceTarget::No,
```

Comment:
Could we please also have a test having a validation method so that we test the ordering?